source/pele/sumgaussianpot.h

```diff
@@ -0,0 +1,120 @@
+#ifndef _PELE_SUMGAUSSIANPOT_H
+#define _PELE_SUMGAUSSIANPOT_H
+
+#include "pele/base_potential.h"
+
+namespace pele {
+    
+class GaussianPot : public BasePotential {
+private:
+    const Array<double> m_mean;
+    const Array<double> m_cov_diag;
+    const double m_gauss_prefactor;
+    const size_t m_bdim;
+    std::vector<double> m_diag_icov;
+public:
+    GaussianPot(Array<double> mean, Array<double> cov_diag)
+        : m_mean(mean.copy()),
+          m_cov_diag(cov_diag.copy()),
+          m_gauss_prefactor(-1),
+          m_bdim(mean.size())
+    {
+        assert(m_mean.size() == m_bdim);
+        assert(m_cov_diag.size() == m_bdim);
+        for (size_t i = 0; i < m_bdim; ++i) {
+            m_diag_icov.push_back(1. / m_cov_diag[i]);
+        }
+        m_diag_icov.swap(m_diag_icov);
```

Comment:
This has been replaced by shrink_to_fit now: http://www.cplusplus.com/reference/vector/vector/shrink_to_fit/